external-crates/move/crates/move-compiler/src/typing/ast.rs

```diff
@@ -172,15 +180,19 @@ pub enum UnannotatedExp_ {
     Vector(Loc, usize, Box<Type>, Box<Exp>),
 
     IfElse(Box<Exp>, Box<Exp>, Box<Exp>),
-    While(Box<Exp>, Box<Exp>),
-    Loop { has_break: bool, body: Box<Exp> },
+    While(Var, Box<Exp>, Box<Exp>),
+    Loop {
+        name: Var,
+        has_break: bool,
+        body: Box<Exp>,
+    },
     Block(Sequence),
     Assign(LValueList, Vec<Option<Type>>, Box<Exp>),
     Mutate(Box<Exp>, Box<Exp>),
     Return(Box<Exp>),
     Abort(Box<Exp>),
-    Break,
-    Continue,
+    Give(Var, Box<Exp>),
```

Comment:
Is this purely a naming convention in the compiler, or does `give` show up in source code?

Reply:
This is internal to the compiler, toward a reusable form that will also help with named blocks and lowering macro control flow.